cmake/test/gtest.cmake

```diff
@@ -442,6 +442,31 @@ else()
   endif()
 endif()
 
+if(NOT GTEST_FOUND)
+  # attempt to find the modern GTest config.
+  find_package(GTest 1.8.1 QUIET CONFIG)
+  if(GTest_FOUND AND TARGET GTest::gtest AND TARGET GTest::gtest_main)
+    set(GTEST_FOUND TRUE)
+    add_library(gtest ALIAS GTest::gtest)
+    add_library(gtest_main ALIAS GTest::gtest_main)
+    set(GTEST_INCLUDE_DIRS)
+    set(GTEST_LIBRARY_DIRS)
+    set(GTEST_LIBRARIES gtest)
+    set(GTEST_MAIN_LIBRARIES gtest_main)
+    set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
+  endif()
+  if(GTest_FOUND AND TARGET GTest::gmock AND TARGET GTest::gmock_main)
+    set(GMOCK_FOUND TRUE)
+    add_library(gmock ALIAS GTest::gmock)
+    add_library(gmock_main ALIAS GTest::gmock_main)
+    set(GMOCK_INCLUDE_DIRS)
+    set(GMOCK_LIBRARY_DIRS)
+    set(GMOCK_LIBRARIES gmock)
+    set(GMOCK_MAIN_LIBRARIES gmock_main)
+    set(GMOCK_BOTH_LIBRARIES ${GMOCK_LIBRARIES} ${GMOCK_MAIN_LIBRARIES})
+  endif()
+endif()
+
```

Comment:
The added logic looks completely "standalone". I don't see why this should be necessary. The CMake code before already calls `find_package(GMock QUIET)` / `find_package(GTest QUIET)`. Why it gtest not found by these?

Anyway in case there needs to be custom logic for Windows it should be integrated into the existing logic and it should be obvious why the existing logic isn't sufficient.

Reply:
Thanks. I added some comments on why explicitly check `GTest` config files would help in some cases. Hope that justifies the reason why it is nice to have it.

Reply:
I still don't see why this block should be standalone / an afterthought below all the existing logic.

Reply:
@seanyen Friendly ping.

Reply:
Thanks for the ping. I didn't have a chance to refactor this logic for now and this is workaround by placing the GTest libs locations like other platforms. I am closing it for now (maybe revisit again later) to reduce the noise.